server/src/main/java/io/druid/client/selector/RandomServerSelectorStrategy.java

```diff
@@ -34,4 +38,14 @@ public QueryableDruidServer pick(Set<QueryableDruidServer> servers, DataSegment
   {
     return Iterators.get(servers.iterator(), random.nextInt(servers.size()));
   }
+
+  @Override
+  public List<QueryableDruidServer> pick(
+      Set<QueryableDruidServer> servers, DataSegment segment, int numServersToPick
+  )
+  {
+    List<QueryableDruidServer> list = Lists.newArrayList(servers);
+    Collections.shuffle(list);
```

Comment:
for repeatability, can you pass `random` into here?

Or alternatively, forget repeatiblity and use thread local random instead of the class's static random